### PR TITLE
Fixed an issue with spreading a spreadable generic expression into generic JSX

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/51577

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -28781,7 +28781,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     spread = getSpreadType(spread, createJsxAttributesType(), attributes.symbol, objectFlags, /*readonly*/ false);
                     attributesTable = createSymbolTable();
                 }
-                const exprType = getReducedType(checkExpressionCached(attributeDecl.expression, checkMode));
+                const exprType = getReducedType(checkExpressionCached(attributeDecl.expression));
```
Comment (Andarist) [inline src/compiler/checker.ts:28784]: this aligns the logic here for the equivalent call in the `checkObjectLiteral`:
https://github.dev/microsoft/TypeScript/blob/4ea8ec49e8aed6601fb6b978fd08031ec058da04/src/compiler/checker.ts#L28541

note that neither of those lines were touched recently

File: tests/baselines/reference/contextuallyTypedStringLiteralsInJsxAttributes02.errors.txt [hunk 2]
```diff
@@ -84,7 +84,7 @@ tests/cases/conformance/types/contextualTypes/jsxAttributes/file.tsx(36,44): err
 !!! error TS2769:     Type '{ extra: true; goTo: string; }' is not assignable to type 'IntrinsicAttributes & ButtonProps'.
 !!! error TS2769:       Property 'extra' does not exist on type 'IntrinsicAttributes & ButtonProps'.
 !!! error TS2769:   Overload 2 of 2, '(linkProps: LinkProps): Element', gave the following error.
-!!! error TS2769:     Type '{ extra: true; goTo: "home"; }' is not assignable to type 'IntrinsicAttributes & LinkProps'.
+!!! error TS2769:     Type '{ extra: true; goTo: string; }' is not assignable to type 'IntrinsicAttributes & LinkProps'.
```
Comment (Andarist) [inline tests/baselines/reference/contextuallyTypedStringLiteralsInJsxAttributes02.errors.txt:87]: At first, I assumed that this is an acceptable change but I rechecked the equivalent call using a call expression and the reported error there doesn't contain a widened string:
[TS playground](https://www.typescriptlang.org/play?ts=5.0.0-dev.20221117#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wChSkAPSWOYAOxiSk3STgGEAbYNAaxQAjLkgAKOMAGc4Ab1JwFcNAAtgXACZF6AfgBccSTCgMA5gG55itFxSTJAORQgkeg0dMWAvuSo14DJhY2OAAhAFcYGAh6cQgpOComenVpbl4BYTEJaTlFOGi0-n0ACj59ACIRTBhyuAAfOHLjE2UagEo4AF4APjgANwhgdS8famh-RmZWNHYAGQY+WPjEpGTUnn4hESWcywUTCAAVCArlXCRahvK0aJh0GpGKMdp1JGsUIjhMMPoMYGi4ABZFAMcKRaLFQQRKIxbL6MEwnZtfQAKQAygANAB0AFERM5GBZfOM4K93p9vr8YP96ECQfQERCePRFnC4PMWUjUZjcfjVjALOQbvRDHBBABmLpwAA8wNB0IBMixypkB2OunKZ2c5U8ngSlCMKDgAHpuoLhaKAO7AGDKDgoLhcKVyhkK+jFGRwZVY1VHE4ELVIfB6xJQFD6IxhdieNpAA)

Gonna have to take another look at this to understand why JSX and a regular call different in behavior for this case after my change.

**Review by weswigham — APPROVED** [review]
Just want to see some extended suites be green and this looks reasonable to me~

**Comment by weswigham** [comment]
@typescript-bot run dt
@typescript-bot test top100

**Comment by jakebailey** [comment]
This PR appears to have broken main via a baseline change.

**Pull request merged**